### PR TITLE
Report all "mutually left recursive rules" errors and fix positions

### LINKED ISSUE #34: Squashed errors for mutually left-recursive rules with incorrect positions
### Grammar

```g4
grammar Test;

a: e;
e: a;

b: c;
c: d;
d: c;

A: A 'B';
C: D;
D: E;
E: D;
```

### Actual

```
error(119): Test.g4::: The following sets of rules are mutually left-recursive [A] and [D, E]
error(119): Test.g4::: The following sets of rules are mutually left-recursive [a, e] and [c, d]
```

### Expected

```
error(119): Test.g4:10:0: The following sets of rules are mutually left-recursive [A]
error(119): Test.g4:12:0: The following sets of rules are mutually left-recursive [D, E]
error(119): Test.g4:3:0: The following sets of rules are mutually left-recursive [a, e]
error(119): Test.g4:7:0: The following sets of rules are mutually left-recursive [c, d]
```

### PR DESCRIPTION
**Pull request by KvanTTT**
*Porting a PR from ANTLR4 repository (https://github.com/antlr/antlr4/pull/3779)*

It fixes https://github.com/antlr/antlr5/issues/34 but is should be moved to the current repo as [I wrote](https://github.com/antlr/antlr5-specs/discussions/12#discussioncomment-8361778).

### CODE REVIEW

File: tool/src/org/antlr/v5/tool/ErrorType.java [hunk 1]
```diff
@@ -417,7 +417,7 @@ public enum ErrorType {
 	 * The following sets of rules are mutually left-recursive
 	 * <em>[rules]</em></p>
 	 */
-	LEFT_RECURSION_CYCLES(119, "The following sets of rules are mutually left-recursive <arg:{c| [<c:{r|<r.name>}; separator=\", \">]}; separator=\" and \">", ErrorSeverity.ERROR),
+	LEFT_RECURSION_CYCLES(119, "The following set of rules are mutually left-recursive [<arg:{r|<r.name>}; separator=\", \">]", ErrorSeverity.ERROR),
```
Comment (ericvergnaud) [inline tool/src/org/antlr/v5/tool/ErrorType.java]: Did you mean "the following rules" ? Please also update the comment ?

**Review by ericvergnaud — CHANGES_REQUESTED** [review]

File: tool/src/org/antlr/v5/tool/ErrorType.java (hunk 1 above)
Comment (KvanTTT) [reply tool/src/org/antlr/v5/tool/ErrorType.java]: Originally it was `sets` that's why I didn't change original semantics. But I can rephrase with `the following rules`, yes, probably it looks a bit more clear for an end user.

**Comment by KvanTTT** [comment]
Done. Could you please also move https://github.com/antlr/antlr5/issues/34 to our antlr5 repository for clarity?

**Comment by ericvergnaud** [comment]
> Done. Could you please also move [antlr/antlr4#3640](https://github.com/antlr/antlr4/issues/3640) to our antlr5 repository for clarity?

Done

**Comment by KvanTTT** [comment]
Thanks, I've fixed the issue number in commit message and now it's ready review.

**Review by ericvergnaud — APPROVED** [review]

**Comment by KvanTTT** [comment]
Please let me merge my own MRs next times.

**Pull request merged**